### ISSUE TITLE
Template extraction on ArrayObject fails if non-templated keys are different

### ISSUE THREAD

**Issue by danog** [vimeo/psalm#8565]
https://psalm.dev/r/21b16158c8

**Comment by orklah** [comment]
Yeah, strange. I thought it could have been because Psalm collapsed the two ArrayObject into a single one(but couldn't with objects) but it doesn't seem so: https://psalm.dev/r/61cd0ec01a
